R/macros.R

```diff
@@ -0,0 +1,99 @@
+#' General Confluence macro builder for internal use
+#' @param type inline or block code style to be used for the HTML content
+#' @param name Confluence macro name
+#' @param parameters named list of optional macro parameters
+#' @param body optional \code{confl-ac-rich-text-body} content
+#' @return HTML
+#' @keywords internal
+conf_macro_generator <- function(type = c('inline', 'block'),
+                                 name, parameters, body) {
+  type <- match.arg(type)
+
+  macro <- switch(
+    type,
+    inline = '`',
+    block = '\n```{=html}\n')
+
+  macro <- glue('{macro}<ac:structured-macro ac:name="{name}">')
+
+  if (!missing(parameters)) {
+    for (parameter in names(parameters)) {
+      macro <- paste0(
+        macro,
+        glue('<ac:parameter ac:name="{parameter}">'),
```

Comment:
```suggestion
        glue::glue('<ac:parameter ac:name="{parameter}">'),
```

Reply:
Why the need to specify the `glue` package here in such an explicit way?

I have not seen a local `glue` function defined in the `conflr` package, and this PR imports `glue` into the `conflr` pkg namespace from `glue` -- so it should just work fine.

No hard preference ... but I have not really used `::` when it was not really required in other packages, so I wonder why you prefer that.

Reply:
Ah, importing glue is fine. There's no opinionated reason that I didn't import it. But..., 

> this PR imports `glue` into the `conflr` pkg namespace from `glue`

maybe you forgot to include the related changes in this PR?

Reply:
> maybe you forgot to include the related changes in this PR?

definitely :man_facepalming: 

just pushed -- but I can revert to `::` as well (or update all the other `glue::glue` cases to drop the explicit pkg reference), just let me know what you prefer with this